### DISCUSSION TITLE
Actions Vs Bulk Actions

### DISCUSSION THREAD

**Discussion by usernotnull** [rappasoft/laravel-livewire-tables#302] [Feature Requests]
I was thinking of writing my own implementation, but this would be great to have natively, depending on how the devs see its usefulness.
Common actions of tables are importing/exporting ALL items. Common Bulk actions are exporting/deleting/custom action on selected rows.
So in theory, there would be 2 kinds of actions, the ones that do not require rows selected, and the "Bulk actions" button would be replaced by "Actions" if no row is selected, and when a row is selected, changed back to "Bulk Actions".
What do you guys think?

**Comment by rappasoft** [comment]
It's an interesting idea, however, I don't know how we would accomplish it in the current version because it would most likely include breaking changes.

**Reply by usernotnull** [reply]
Bulk Actions itself is an optional trait, I was assuming a mere copy/paste + refactoring name to "WithActions" would work.
The @if in blade already is flexible enough.
